addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3380,9 +3380,21 @@ export class OdooEditor extends EventTarget {
         if (!rangeContent.hasChildNodes()) {
             return;
         }
+        // find list of ancestors.
+        const ancestorList = range.commonAncestorContainer.nodeType === Node.ELEMENT_NODE ?
+                            ancestors(range.commonAncestorContainer.firstChild) : ancestors(range.commonAncestorContainer);
+        const closestLi = closestElement(selection.anchorNode, 'li');
+        for (const ancestor of ancestorList) {
+            if ((closestLi && rangeContent.firstChild.nodeName === 'LI') || isBlock(rangeContent.firstChild)) {
+                break;
+            }
```

Comment:
I don't understand the use case for this `break` in case of `LI`. Like I said above I think this would benefit from a comment to explain why this is a special case that needs to be handled separately.

Reply:
@dmo-odoo 
Here, The ancestorList contains the list of elements up to the html.
we only need the ancestorList up to li or block level Element, so to stop the iteration of the ancestor list I have added that condition and break the iteration after rangeContent is wrapped in either li or block element.

After your above suggestion, I have already removed this condition from here, and now I am finding ancestorList up to the li or block element instead html.
I think it's quite a significant change in code so, I am moving this task to RD Testing IN for testing.